milestones/data.py

```diff
@@ -0,0 +1,390 @@
+# pylint: disable=no-member
+"""
+Application data management/abstraction layer.  Responsible for:
+
+1) Accessing information from various resources:
+* Internal application state  (through local models in models.py)
+* External application state  (ORM bindings with other apps, yuck)
+* Remote data services (through service adapters in resources.py)
+
+2) Calculating derivative information from existing state:
+* Algorithms and data manipulations
+* Aggregations
+* Annotations
+* Alternative data representations
+
+Accepts and returns standard Python data structures (dicts, arrays of dicts)
+for easy consumption and manipulation by callers -- the queryset stops here!
+
+Note the terminology difference at this layer vs. API -- create/fetch/update/delete/
+
+When the time comes for remote resources, import the module like so:
+if getattr(settings, 'TEST_MODE', False) or os.getenv('TRAVIS_MODE', False):
+    import milestones.tests.mocks.resources as remote
+else:
+    import milestones.resources as remote
+"""
+from . import exceptions
+from . import models as internal
+from . import serializers
+
+RELATIONSHIP_TYPES = {
+    'FULFILLS': "fulfills",
+    'REQUIRES': "requires",
+}
```

Comment:
I believe these are meant to be system defined string constants.

I don't think we have a consistent best practice for constants in Open edX. Looking around the interwebs, this looks interesting:

http://code.activestate.com/recipes/65207-constants-in-python/?in=user-97991